app/merginapi.cpp

```diff
@@ -111,10 +111,30 @@ void MerginApi::uploadProject(QString projectName)
     }
 }
 
+void MerginApi::authorize(QString username, QString password)
+{
+    QString concatenated = username + ":" + password;
+    mToken = concatenated.toLocal8Bit().toBase64();
+    mUsername = username;
+    emit usernameChanged();
+    emit authChanged();
+}
+
+void MerginApi::logoutRequested()
```

Comment:
let's rename this to something like clearAuth()  ... logoutRequested sounds like a signal (similar to authRequested)